### PR TITLE
gitlab_project: Add a parameter to specify Git LFS

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

* Update docs
* Update example
* Refactor code
* Add a parameter ``lfs_enabled`` to specify Git Large file systems

Fixes: #1506

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
changelogs/fragments/1506_gitlab_project.yml
plugins/modules/source_control/gitlab/gitlab_project.py

### CODE REVIEW

**Comment by paroque28** [comment]
It works:

```
ok: [localhost] => (item=anongit.freedesktop.org.pkg-config) => {
    "ansible_loop_var": "item",
    "changed": false,
    "invocation": {
        "module_args": {
            "api_password": null,
            "api_token": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "api_url": ****************,
            "api_username": null,
            "description": null,
            "group": **************,
            "import_url": null,
            "issues_enabled": true,
            "lfs_enabled": false,
            "merge_method": "rebase_merge",
            "merge_requests_enabled": true,
            "name": "anongit.freedesktop.org.pkg-config",
            "path": null,
            "snippets_enabled": true,
            "state": "present",
            "validate_certs": true,
            "visibility": "public",
            "wiki_enabled": true
        }
    },
    "item": "anongit.freedesktop.org.pkg-config",
    "msg": "No need to update the project anongit.freedesktop.org.pkg-config",
    "project": {
        "_links": {
            **************
        },
        "allow_merge_on_skipped_pipeline": null,
        "archived": false,
        "auto_cancel_pending_pipelines": "enabled",
        "auto_devops_deploy_strategy": "continuous",
        "auto_devops_enabled": false,
        "autoclose_referenced_issues": true,
        "avatar_url": null,
        "build_coverage_regex": null,
        "build_git_strategy": "fetch",
        "build_timeout": 3600,
        "builds_access_level": "enabled",
        "can_create_merge_request_in": true,
        "ci_config_path": "",
        "ci_default_git_depth": 50,
        "container_expiration_policy": {
            "cadence": "1d",
            "enabled": true,
            "keep_n": 10,
            "name_regex": null,
            "name_regex_keep": null,
            "next_run_at": "2020-12-24T16:50:38.787Z",
            "older_than": "90d"
        },
        "container_registry_enabled": true,
        "created_at": "2020-12-22T16:44:43.995Z",
        "creator_id": 135,
        "default_branch": "master",
        "description": null,
        "emails_disabled": null,
        "empty_repo": false,
        "forking_access_level": "enabled",
        "forks_count": 0,
        "http_url_to_repo": ****************,
        "id": 2025,
        "import_error": null,
        "import_status": "none",
        "issues_access_level": "enabled",
        "issues_enabled": true,
        "jobs_enabled": true,
        "last_activity_at": "2020-12-22T16:44:43.995Z",
        "lfs_enabled": false,
        "merge_method": "rebase_merge",
        "merge_requests_access_level": "enabled",
        "merge_requests_enabled": true,
        "name": "anongit.freedesktop.org.pkg-config",
        "name_with_namespace": ***********,
        "namespace": {
            "avatar_url": null,
            "full_path": **********,
            "id": 261,
            "kind": "group",
            "name": "WRLinux-CD-Core",
            "parent_id": 260,
            "path": "WRLinux-CD-Core",
            "web_url": ********
        },
        "only_allow_merge_if_all_discussions_are_resolved": false,
        "only_allow_merge_if_pipeline_succeeds": false,
        "open_issues_count": 0,
        "packages_enabled": true,
        "pages_access_level": "private",
        "path": "anongit.freedesktop.org.pkg-config",
        "path_with_namespace": **********,
        "permissions": {
            "group_access": {
                "access_level": 50,
                "notification_level": 3
            },
            "project_access": null
        },
        "printing_merge_request_link_enabled": true,
        "public_jobs": true,
        "readme_url":***********,
        "remove_source_branch_after_merge": true,
        "repository_access_level": "enabled",
        "request_access_enabled": true,
        "resolve_outdated_diff_discussions": false,
        "runners_token": *********,
        "service_desk_address": null,
        "service_desk_enabled": false,
        "shared_runners_enabled": true,
        "shared_with_groups": [],
        "snippets_access_level": "enabled",
        "snippets_enabled": true,
        "ssh_url_to_repo": ******,
        "star_count": 0,
        "suggestion_commit_message": null,
        "tag_list": [],
        "visibility": "public",
        "web_url": **********,
        "wiki_access_level": "enabled",
        "wiki_enabled": true
    }
}
```

File: plugins/modules/source_control/gitlab/gitlab_project.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 short_description: Creates/updates/deletes GitLab Projects
 description:
   - When the project does not exist in GitLab, it will be created.
-  - When the project does exists and state=absent, the project will be deleted.
+  - When the project does exists and C(state=absent), the project will be deleted.
```
Comment (felixfontein) [inline plugins/modules/source_control/gitlab/gitlab_project.py:17]:
```suggestion
  - When the project does exists and I(state=absent), the project will be deleted.
```

File: plugins/modules/source_control/gitlab/gitlab_project.py [hunk 2]
```diff
@@ -63,20 +63,18 @@
...
     type: bool
     default: yes
   visibility:
     description:
-      - Private. Project access must be granted explicitly for each user.
-      - Internal. The project can be cloned by any logged in user.
-      - Public. The project can be cloned without any authentication.
+      - I(Private) Project access must be granted explicitly for each user.
+      - I(Internal) The project can be cloned by any logged in user.
+      - I(Public) The project can be cloned without any authentication.
```
Comment (felixfontein) [inline plugins/modules/source_control/gitlab/gitlab_project.py:77]:
```suggestion
      - C(private) Project access must be granted explicitly for each user.
      - C(internal) The project can be cloned by any logged in user.
      - C(public) The project can be cloned without any authentication.
```

File: plugins/modules/source_control/gitlab/gitlab_project.py [hunk 3]
```diff
@@ -103,9 +101,25 @@
...
     default: merge
     version_added: "1.0.0"
+  lfs_enabled:
+    description:
+      - Enable Git large file systems to manages large files such
+        as audio, video, and graphics files.
+    type: bool
+    required: false
+    default: false
+    version_added: "1.4.0"
```
Comment (felixfontein) [inline plugins/modules/source_control/gitlab/gitlab_project.py:111]:
```suggestion
    version_added: "2.0.0"
```

File: plugins/modules/source_control/gitlab/gitlab_project.py [hunk 4]
```diff
@@ -103,9 +101,25 @@
...
 '''
 
 EXAMPLES = r'''
+- name: Create GitLab Project
+  community.general.gitlab_project:
+    api_url: https://gitlab.example.com/
+    api_token: "{{ api_token }}"
+    validate_certs: False
```
Comment (felixfontein) [inline plugins/modules/source_control/gitlab/gitlab_project.py:119]:
```suggestion
```
Would be nice if the first example does **not** specify unsafe defaults :)

File: plugins/modules/source_control/gitlab/gitlab_project.py [hunk 5]
```diff
@@ -63,20 +63,18 @@
...
     type: bool
     default: yes
   visibility:
     description:
-      - Private. Project access must be granted explicitly for each user.
-      - Internal. The project can be cloned by any logged in user.
-      - Public. The project can be cloned without any authentication.
+      - C(Private) Project access must be granted explicitly for each user.
+      - C(Internal) The project can be cloned by any logged in user.
+      - C(Public) The project can be cloned without any authentication.
```
Comment (felixfontein) [inline plugins/modules/source_control/gitlab/gitlab_project.py:77]:
```suggestion
      - C(private) Project access must be granted explicitly for each user.
      - C(internal) The project can be cloned by any logged in user.
      - C(public) The project can be cloned without any authentication.
```
The values are lower-case according to the `choices` below.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@Akasurde thanks for implementing this!
@paroque28 thanks for testing!

**Comment by Akasurde** [comment]
@felixfontein Thanks for the review and merge.

**Pull request merged**